opsdroid/connector/gitter/__init__.py

```diff
@@ -47,28 +46,28 @@ def build_url(self,base_url , *res, **params):
 
   async def listen(self):
     _LOGGER.debug("Listening with gitter stream")
-    while True:
+    while self.listening:
       await asyncio.sleep(self.update_interval)
       async for data in self.response.content.iter_chunked(1024):
-        _LOGGER.info(data)
-        try:
-            data = await self.parse_message(data)
-            if data !=None:
-              message = Message(
-              data["text"],
-              data["fromUser"]["username"],
-              self.room_id,
-              self)
-              await self.opsdroid.parse(message)
-        except Exception as err:
-          _LOGGER.error("Unable to parse message %s", data)
-          _LOGGER.error(err.with_traceback())
+        message = await self.parse_message(data)
+        if message !=None:
+          await self.opsdroid.parse(message)
 
   async def parse_message(self,message):
     message = message.decode('utf-8').rstrip("\r\n")
     if len(message) > 1:
       message = json.loads(message)
-      return message
+      _LOGGER.debug(message)
+      try:
+        return Message(
+            message["text"],
+            message["fromUser"]["username"],
+            self.room_id,
+            self)
+      except Exception as err:
```

Comment:
This is a very broad exception catch. Is it possible to narrow this down a little? I would expect you are only likely to hit `KeyError` given the contents of the try.